plans/subplans/provision.pp

```diff
@@ -76,30 +90,34 @@
 #   net-ssh library
 #
 plan pecdm::subplans::provision(
-  Enum['xlarge', 'large', 'standard']           $architecture         = 'standard',
-  Enum['development', 'production', 'user']     $cluster_profile      = 'development',
-  Integer                                       $compiler_count       = 1,
-  Optional[String[1]]                           $ssh_pub_key_file     = undef,
-  Optional[Integer]                             $node_count           = undef,
-  Optional[Variant[String[1],Hash]]             $instance_image       = undef,
-  Optional[Variant[String[1],Array[String[1]]]] $subnet               = undef,
-  Optional[String[1]]                           $subnet_project       = undef,
-  Optional[Boolean]                             $disable_lb           = undef,
-  Enum['private', 'public']                     $ssh_ip_mode          = 'public',
-  Enum['private', 'public']                     $lb_ip_mode           = 'private',
-  Array                                         $firewall_allow       = [],
-  Hash                                          $extra_terraform_vars = {},
-  Boolean                                       $replica              = false,
-  Boolean                                       $native_ssh           = false,
+  Enum['xlarge', 'large', 'standard']           $architecture           = 'standard',
+  Enum['development', 'production', 'user']     $cluster_profile        = 'development',
+  Integer                                       $compiler_count         = 1,
+  Optional[String[1]]                           $ssh_pub_key_file       = undef,
+  Optional[Integer]                             $node_count             = undef,
+  Optional[Variant[String[1],Hash]]             $instance_image         = undef,
+  Optional[Integer]                             $windows_node_count     = undef,
+  Optional[Variant[String[1],Hash]]             $windows_instance_image = undef,
+  Optional[String[1]]                           $windows_user           = 'windows',
+  Sensitive[String[1]]                          $windows_password,
```

Comment:
Needs to be set to optional with default undef or else it'll be required even if you're not deploying windows nodes